test/k8sT/Services.go

```diff
@@ -816,25 +816,25 @@ var _ = SkipDescribeIf(helpers.RunsOn54Kernel, "K8sServicesTest", func() {
 			}
 
 			var wg sync.WaitGroup
-			wg.Add(len(testCases) * 2)
-			for _, testCase := range testCases {
-				for _, name := range []string{be1Name, be2Name} {
-					go func(tc lrpTestCase, want string) {
+			for _, tc := range testCases {
+				pods, err := kubectl.GetPodNames(helpers.DefaultNamespace, tc.selector)
+				Expect(len(pods)).Should(BeNumerically(">", 0), "no pod exists by filter %q", tc.selector)
+				Expect(err).Should(BeNil(), "cannot retrieve pod names by filter %q", tc.selector)
+				for _, pod := range pods {
+					wg.Add(1)
+					go func(tc lrpTestCase, pod string) {
 						defer GinkgoRecover()
 						defer wg.Done()
 						Eventually(func() bool {
-							pods, err := kubectl.GetPodNames(helpers.DefaultNamespace, tc.selector)
-							Expect(err).Should(BeNil(), "cannot retrieve pod names by filter %q", tc.selector)
-							Expect(len(pods)).Should(BeNumerically(">", 0), "no pod exists by filter %q", tc.selector)
 							ret := true
-							for _, pod := range pods {
+							for _, want := range []string{be1Name, be2Name} {
 								res := kubectl.ExecPodCmd(helpers.DefaultNamespace, pod, tc.cmd)
 								Expect(err).To(BeNil(), "%s failed in %s pod", tc.cmd, pod)
```

Comment:
should `err` be `res`?

Reply:
I didn't touch this code. But, yeah, the error check doesn't seem right. Good catch!